updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -0,0 +1,357 @@
+# frozen_string_literal: true
+
+module Dependabot
+  class Updater
+    module Operations
+      class UpdateAllVersions
+        def self.applies_to?(job:)
+          return false if job.security_updates_only?
+          return false if job.updating_a_pull_request?
+          return false if job.dependencies&.any?
+
+          true
```

Comment:
I did think about using an experiment to roll this out, but since it is just extracting code and clipping off branches of if statements that are essentially `false` when the `applies_to?` method is true, I'm of fairly high confidence this is a refactor with no expected behaviour change.

I think smoke testing + staging is sufficient to roll into production and watch the first few jobs run vs a gradual rollout via feature flag - the gradual rollout might actually give us false confidence as any problems may not be statistically significant in a sample of 10% or 25% of jobs.